hq/app/logic/Retry.scala

```diff
@@ -0,0 +1,24 @@
+package logic
+
+import utils.attempt.{Attempt, Failure}
+
+import scala.concurrent.ExecutionContext
+object Retry {
+
+  def until[A](body: => Attempt[A], predicate: A => Boolean, failureMessage: String, maxAttempt: Int = 10)(implicit ec: ExecutionContext): Attempt[A] = {
```

Comment:
Very minor point, but it's a Scala convention to make functions require parentheses when they perform side-effects. This would just make it explicit that we're planning to do IO here. Parens-less pure access is fine, but we know `body` will go to AWS.

Reply:
i think putting `()` will make `until` usage a bit uglier  as i have to use it like `{ () => IAMClient.generateCredentialsReport(client) }`.  
Retry is also not bound to AWS as it's generic at the moment. I think the convention around parentheses cares about type function declarations. (e.g. `def foo ` vs `def foo()`) . There is an example below.
https://docs.scala-lang.org/style/naming-conventions.html#parentheses